test/unittests/LorisForms_Test.php

```diff
@@ -262,6 +470,1788 @@ function testAddElementPassword()
+            "abc_default", $this->form->staticHTML($this->form->form["abc"])
+        );
+    }
+
+    /**
+     * Test that dateHTML returns the correctly formatted HTML 
+     * when no attributes or options are specified
+     *
+     * @covers LorisForm::dateHTML
+     * @return void
+     */
+    function testDateHTMLWithNoOptions()
+    {
+        $this->form->addDate("abc", "Hello", array());
+        //The string below is written this way to stay within the 85 char. limit
+        $this->assertEquals(
+            "<input name=\"abc\" type=\"date\"".
+                "    onChange=\"this.setCustomValidity('')\" >", 
+            $this->form->dateHTML($this->form->form["abc"])
+        );
+    } 
+
+    /**
+     * Test that dateHTML returns the correctly formatted HTML 
+     * with the given attributes and options. This returns the HTML when 'format' 
+     * is set to either 'my' or 'ym' in the options array
+     *
+     * @covers LorisForm::dateHTML
+     * @return void
+     */
+    function testDateHTMLWithFormatNotSetToY()
+    {
+        $testAttributes = array('class' => 'class1',
+                                'disabled' => 'yes',
+                                'requireMsg' => 'requireMsg1',
+                                'required' => 'required1');
+        $testOptions = array('minYear' => '2010',
+                             'maxYear' => '2019',
+                             'format' => 'ym');
+
+        $this->form->addDate("abc", "Hello", $testOptions, $testAttributes);
+        $this->form->setDefaults(array('abc' => 'abc_default'));
+        //The string below is written this way to stay within the 85 character limit
+        $this->assertEquals(
+            "<input name=\"abc\" type=\"month\" class=\"class1\"".
+                " min=\"2010-01\" max=\"2019-12\" onChange=".
+                "\"this.setCustomValidity('')\" disabled  value=\"abc_default-\" >",
+            $this->form->dateHTML($this->form->form["abc"])
+        );
+    }
+
+    /**
+     * Test that dateHTML calls the yearHTML function 
+     * when 'format' is set to 'y' in the options array
+     *
+     * @covers LorisForm::dateHTML
+     * @return void
+     */
+    function testDateHTMLWithFormatSetToY()
+    {
+        $testAttributes = array('class' => 'class1',
+                                'disabled' => 'yes',
+                                'requireMsg' => 'requireMsg1',
+                                'required' => 'required1');
+        $testOptions = array('minYear' => '2010',
+                             'maxYear' => '2019',
+                             'format' => 'y');
+        $this->form = $this->getMockBuilder('LorisForm')
+            ->setMethods(array('yearHTML'))
+            ->getMock();
+        $this->form->expects($this->once())
+            ->method('yearHTML');
+        $this->form->addDate("abc", "Hello", $testOptions, $testAttributes);
+        $this->form->dateHTML($this->form->form["abc"]);
+    }
```

Comment:
Should this have an assertEquals?

Reply:
Yes it should! I will add this now.

Reply:
Actually @PapillonMcGill after looking at it again, it does not. This test is just checking that if the 'format' option (in the testOptions array) is set to 'y' (meaning year), it will call the 'yearHTML' method.

Reply:
Then why is the function just over it (testDateHTMLWithFormatNotSetToY()) check that the date entered is within range?

Reply:
Because if you look at the [yearHTML](https://github.com/aces/Loris/blob/2e3278658d806e6ee104c06de0e005c3a4c4c864/php/libraries/LorisForm.class.inc#L1077) method, it uses Smarty_neurodb and it's not mocking correctly.

Reply:
@AlexandraLivadas 
Good job, you are almost there. there is no need to continue exploring the yearHTML method. The method you need to test is whether it calls the yearHTML method. 
        $this->form->expects($this->once())
             ->method('yearHTML')
             ->willReturn('called yearHTML function');
        $this->form->addDate("abc", "Hello", $testOptions, $testAttributes);
        $this->assertEquals("called yearHTML function",$this->form->dateHTML($this->form->form["abc"]));